qiskit/algorithms/eigensolvers/ssvqe.py

```diff
@@ -0,0 +1,591 @@
+    spin-magnetization quantum number.
+
+    The following attributes can be set via the initializer but can also be read and
+    updated once the SSVQE object has been constructed.
+
+    Attributes:
+            estimator (BaseEstimator): The primitive instance used to perform the expectation
+                estimation of observables.
+            k (int): The number of eigenstates that SSVQE will attempt to find.
+            ansatz (QuantumCircuit): A parameterized circuit used as an ansatz for the
+                wave function.
+            optimizer (Optimizer): A classical optimizer, which can be either a Qiskit optimizer
+                or a callable that takes an array as input and returns a Qiskit or SciPy optimization
+                result.
+            gradient (BaseEstimatorGradient | None): An optional estimator gradient to be used with the
+                optimizer.
+            initial_states (Sequence[QuantumCircuit]): An optional list of mutually orthogonal
+                initial states. If ``None``, then SSVQE will set these to be a list of mutually
+                orthogonal computational basis states.
+            weight_vector (Sequence[float]): A 1D array of real positive-valued numbers to assign
+                as weights to each of the expectation values. If ``None``, then SSVQE will default
+                to [k, k-1, ..., 1].
+            callback (Callable[[int, np.ndarray, Sequence[float], dict[str, Any]], None] | None): A
+                function that can access the intermediate data at each optimization step. These data are
+                the evaluation count, the optimizer parameters for the ansatz, the evaluated mean
+                energies, and the metadata dictionary.
+            check_input_states_orthogonality: A boolean that sets whether or not to check
+                that the value of initial_states passed consists of a mutually orthogonal
+                set of states. If ``True``, then SSVQE will check that these states are mutually
+                orthogonal and return an error if they are not. This is set to ``True`` by default,
+                but setting this to ``False`` may be desirable for larger numbers of qubits to avoid
+                exponentially large computational overhead before the simulation even starts.
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        k: int | None = 2,
+        ansatz: QuantumCircuit | None = None,
+        optimizer: Optimizer | Minimizer = None,
+        initial_point: Sequence[float] = None,
+        initial_states: list[QuantumCircuit] = None,
+        weight_vector: Sequence[float] | Sequence[int] = None,
+        gradient: BaseEstimatorGradient | None = None,
+        callback: Callable[[int, np.ndarray, Sequence[float], float], None] | None = None,
+        check_input_states_orthogonality: bool = True,
+    ) -> None:
+        """
+        Args:
```

Comment:
We could also give a code example somewhere in the docs here, just like in the release notes.

Reply:
Okay, I will work on adding this. I am also considering re-writing the introduction docstring to be more concise and to the point, rather than being a few large blocks of texts. (Similar in style to what the current docstring for VQE with primitives.)